poc/vidpf.py

```diff
@@ -227,34 +229,29 @@ def eval(self,
         # Implementation note: we can save computation by storing
         # `prefix_tree_share` across `eval()` calls for the same report.
         prefix_tree_share: dict[PrefixTreeIndex, PrefixTreeEntry] = {}
+        root = PrefixTreeEntry.root(key, Field2(agg_id))
         proof = PROOF_INIT
-        for prefix in prefixes:
-            if prefix not in range(2 ** (level+1)):
-                raise ValueError("prefix too long")
-
-            seed = key
-            ctrl = Field2(agg_id)
-            for i in range(level+1):
-                node = prefix >> (level - i)
-                idx = PrefixTreeIndex(node, i)
-                for inner_idx in [idx, idx.sibling()]:
-                    # Compute the value for the node and its sibling. The
+        for i in range(level+1):
+            for prefix in prefixes:
+                if prefix not in range(2 ** (level+1)):
+                    raise ValueError("prefix too long")
+
+                # Compute the entry for `prefix`. To do so, we first need to
+                # look up the parent node.
+                idx = PrefixTreeIndex(prefix >> (level - i + 1), i - 1)
```

Comment:
I'm a bit confused with the `+ 1` and `i - 1`, could you explain it in a comment?

Reply:
Done.